shell/src/main/java/org/apache/zeppelin/shell/TerminalInterpreter.java

```diff
@@ -123,6 +129,13 @@ public InterpreterResult internalInterpret(String cmd, InterpreterContext contex
       }
     }
     setParagraphConfig();
+    Properties properties = getProperties();
+    String strIpMapping = properties.getProperty("zeppelin.terminal.ip.mapping");
+    if (!StringUtils.isEmpty(strIpMapping)) {
+      Gson gson = new Gson();
```

Comment:
gson could be a static variable so that we don't need to create it for each call.